### PR TITLE
TE-742 `Header` search bar options

### PR DESCRIPTION
**Pull request by marcdavi-es**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-742)

### What **one** thing does this PR do?
Renders `SearchBar` only if the relevant options props are defined and populated.

### CODE REVIEW

File: src/components/collections/Header/Readme.md [hunk 1]
```diff
@@ -98,6 +98,25 @@ const logoSrc = require('./mock-data/livingstoneLogo.png');
...
+The header can display a clickable icon to reveal the search bar on mobile screens
+
+```jsx
+const { navigationItems } = require('./mock-data/navigationItems');
+const searchBarGuestsOptions = [{ text: '1', value: 1 }];
+const searchBarLocationOptions = [{ text: 'Catania', value: 'catania' }];
+
+<div style={{ backgroundColor: 'grey'}}>
+  <Header
+    logoText={'Livingstone Cottage'}
```
Comment (stephenkr) [inline src/components/collections/Header/Readme.md]: Why not just `logoText='Livingstone Cottage'`?

File: src/components/collections/Header/utils/hasSearchBarOptions.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { size } from 'lodash';
+
+/**
+ * @param  {Object[]|undefined}  guestsOptions
+ * @param  {Object[]|undefined}  locationOptions
+ * @return {Boolean}
+ */
+export const hasSearchBarOptions = (guestsOptions, locationOptions) =>
+  !!size(guestsOptions) && !!size(locationOptions);
```
Comment (stephenkr) [inline src/components/collections/Header/utils/hasSearchBarOptions.js]: I think its clearer to do a numeric comparision such as `size(guestsOptions) > 0 && size(locationOptions) > 0`

File: src/components/collections/Header/Readme.md (hunk 1 above)
Comment (marcdavi-es) [reply src/components/collections/Header/Readme.md]: Yep - just a lazy copy and paste. I'll fix the whole file.

**Comment by marcdavi-es** [comment]
@stephenkr Those changes are done.

**Pull request merged**